packages/core/src/internal/clients/vector/IVectorIndexDataClient.ts

```diff
@@ -26,4 +28,12 @@ export interface IVectorIndexDataClient {
     indexName: string,
     ids: Array<string>
   ): Promise<VectorDeleteItemBatch.Response>;
+  getItemBatch(
+    indexName: string,
+    ids: Array<string>
+  ): Promise<VectorGetItemBatch.Response>;
+  getItemMetadataBatch(
+    indexName: string,
+    ids: Array<string>
+  ): Promise<VectorGetItemMetadataBatch.Response>;
```

Comment:
just out of curiosity - is it common for users to only want one or the other of these two types of data? (i.e. they either want the item, or they want the metadata, but they don't want both?)

I can imagine that there might be some server-side implementation / performance details that make it desirable to have these as two separate APIs but I'm not familiar enough with use cases to know if users are going to end up usually only calling one of the 2 of these, or if they will frequently be calling both of them. If the latter, just curious about thoughts about possibly returning both from a single api.

Reply:
A customer asked for these. I am unclear on the particular use outside of the search.

That said, there are two RPCs under the hood: `get item batch` and `get item metadata batch`. Each one has a `metadata selector` that allows us (users) to choose all metadata, some metadata, or no metadata. For the first cut I kept it simple and did not plumb that through just yet.

So in the future if someone only wants the vectors, we can plumb through the `metadata selector` and they can invoke `get item batch with no metadata`. Or we can furnish a new method that is just "get vector".